### PR TITLE
Add Migration Guide for Event Hubs

### PR DESCRIPTION
**Pull request by conniey**
* Adds migration guide for Event Hubs Track 1 -> Track 2
* Fixes typo in migration guide for storage
* Adds .gitignore for .vscode/.factorypath

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    .buildAsyncProducerClient();
+
+CreateBatchOptions options = new CreateBatchOptions()
+    .setPartitionId("my-partition-id");
+
+Mono<Void> sendOperation = producer.createBatch(options).flatMap(batch -> {
+    for (EventData event : data) {
+        // Assuming all events fit into a single batch. This returns false if it does not.
+        // If that is the case, we'll send the full batch then create another one to continue adding events to.
+        batch.tryAdd(event);
```
Comment (richardpark-msft) [inline sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md]: For consistency would this also have a "here's the failure case block" style to it (like the V3 block above?)

```java
if (!batch.tryAdd(event)){
   // Event is too large for batch
}
```

File: sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+### Migrating code from `PartitionReceiver` to `EventHubConsumerClient` for receiving events
+
+In V2, event handlers were passed as positional arguments to `receive`.
+
+In V5, event handlers are passed as part of a `SubscriptionEventHandlers` shaped object.
+
+For example, this code which receives from a partition in V2:
+
+```typescript
```
Comment (jsquire) [inline sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md]: nit:  I think these are meant to be `java`, no?   Same for line 183.

Comment (jsquire) [reply sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md]: Maybe not.....   but that's a bit weird.  Is this section correct?

File: sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+sendOperation.block();
+```
+
+### Migrating code from `PartitionReceiver` to `EventHubConsumerClient` for receiving events
+
+In V2, event handlers were passed as positional arguments to `receive`.
+
+In V5, event handlers are passed as part of a `SubscriptionEventHandlers` shaped object.
+
+For example, this code which receives from a partition in V2:
```
Comment (jsquire) [inline sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md]: You refer to V3 everywhere else.   Was v2 a mistake here?

File: sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+];
+
+const client = EventHubClient.createFromConnectionString(connectionString);
+
+// Would fail if the total size of events exceed the max size supported by the library.
+await client.sendBatch(eventsToSend, partitionId);
+```
+
+In V5:
+```typescript
```
Comment (jsquire) [inline sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md]: nit:  I think you mean `java` here, but the snippet looks like typescript.   Is that intentional?

File: sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+
+## General changes
+
+In the interest of simplifying the API surface, we've made three clients, each with an asynchronous and synchronous
+variant. One client for sending events, `EventHubProducerAsyncClient`, and two for receiving events.
+`EventProcessorClient` is the production level consumer and `EventHubConsumerAsyncClient` for exploration and
+lower-level control of `EventData` consumption.
+
+[EventProcessorClient][EventProcessorClient] supports checkpointing and load balancing through a plugin model.
+Currently, only Azure Blob storage is supported through
```
[the remark this answers is not in the document]
Comment (conniey) [reply sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md]: Thanks for letting me know. I've updated all instances of "Azure blob storage"

File: sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md (hunk 2 above)
Comment (conniey) [reply sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md]: Now it makes sense, I forgot to actually push my latest version.. so some of these comments didn't add up. LOL. 🤣
Connie: "Ctrl+F. Where is this code snippet?!!"

**Pull request merged**